src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,255 +42,137 @@
 
 #include "base/logger.h"
 
+using namespace std::literals;
+
 namespace
 {
-    class IPv4Parser
+    struct IPFilterEntry
     {
-    public:
-        bool tryParse(const char *str)
-        {
-            unsigned char octetIndex = 0;
+        std::string_view range;
+        int accessLevel = 0;
+        // We don't care about this
+        // std::string_view description;
+    };
 
-            const char *octetStart = str;
-            char *endptr;
-            for (; *str; ++str)
-            {
-                if (*str == '.')
-                {
-                    const long int extractedNum = strtol(octetStart, &endptr, 10);
-                    if ((extractedNum >= 0L) && (extractedNum <= 255L))
-                        m_buf[octetIndex++] = static_cast<unsigned char>(extractedNum);
-                    else
-                        return false;
-
-                    if (endptr != str)
-                        return false;
-                    if (octetIndex == 4)
-                        return true;
-
-                    octetStart = str + 1;
-                }
-            }
+    IPFilterEntry parseDATLine(std::string_view line)
+    {
+        IPFilterEntry entry;
 
-            if (str != octetStart)
-            {
-                const long int extractedNum = strtol(octetStart, &endptr, 10);
-                if ((extractedNum >= 0L) && (extractedNum <= 255L))
-                    m_buf[octetIndex] = static_cast<unsigned char>(strtol(octetStart, &endptr, 10));
-                else
-                    return false;
-
-                if ((endptr == str) && (octetIndex == 3))
-                    return true;
-            }
+        // Each line should follow this format:
+        // 001.009.096.105 - 001.009.096.105 , 000 , Some organization
```

Comment:
Either this line describe format incorrectly or you don't take into account spaces so they go into parsed items, e.g. `entry.range` becomes `001.009.096.105 - 001.009.096.105 ` (with space in the end).
The same is about  `parseRange()`.

Reply:
>becomes 001.009.096.105 - 001.009.096.105 (with space in the end).

`std::from_chars()` will stop at the first non-numeric character. So we don't care about trailing spaces.

>The same is about parseRange().

But we care about leading spaces and that function doesn't account for that.